### DISCUSSION TITLE
issue running mips SE(syscall not implemented)

### DISCUSSION THREAD

**Discussion by pedromlabio** [gem5/gem5#1281] [Q&A]
i'm trying to run a mips binary compiled from c, however i get the following error: fatal: syscall set_tid_address (#4252) unimplemented. i have tried multiple ways to avoid this error but i'm very unexperienced in this area, also running an FS simulation is out of question for me since i have no idea how i'd even do that, if anyone knows a way i could either implement this syscall or a way to avoid this error i'd be extremely grateful.

**Comment by LujhCoconut** [comment]
As far as I know, in SE mode, errors related to syscalls are common, especially when gem5 reports  "fatal: syscall xxx(#xxx) unimplemented." This is because SE mode is not designed to fully support complex operating system mechanisms, including many syscalls. These syscalls are typically necessary when running in a complete operating system environment. Therefore, if a program relies on these syscalls for execution, it may fail in SE mode.

In some cases, if only a few syscalls are missing, you can attempt to manually implement these unimplemented calls in gem5.

You can refer to other parts of the code in the `src/arch/mips/isa.cc` and `src/arch/mips/isa.hh` to implement the missing system calls. However, I think this is not an easy work. 

So, it’s advisable to first understand what the missing syscall does. You might refer to how this syscall is implemented in some open-source MIPS projects, or consult AI resources. Finally, you can make targeted modifications based on the isa files in gem5. 

I hope this helps, and good luck!
